hooks/migrate_csi_endpoint.sh

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env bash
+#
+# Copyright 2024 Flant JSC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source /deckhouse/shell_lib.sh
+
+get_config() {
+  echo '{"configVersion":"v1", "beforeHelm": 10}'
+}
+
+run_trigger() {
+  export old_driver_name="linstor.csi.linbit.com"
+  export new_driver_name="replicated.csi.storage.deckhouse.io"
+  export old_attacher="linstor-csi-linbit-com"
+  export new_attacher="replicated-csi-storage-deckhouse-io"
+  export LABEL_KEY="storage.deckhouse.io/need-kubelet-restart"
+  export LABEL_VALUE=""
+  export SECRET_NAME="csi-migration-finished"
+  export NAMESPACE="d8-sds-replicated-volume"
+  export timestamp="$(date +"%Y%m%d%H%M%S")"
+  export affected_pvs_hash=""
+  export WEBHOOK_NAME="d8-sds-replicated-volume-sc-validation"
+  echo "Migration csi shell hook started"
+
+  if kubectl get secret "$SECRET_NAME" -n "$NAMESPACE" > /dev/null 2>&1; then
+    echo "Secret ${NAMESPACE}/${SECRET_NAME} exists. Migration has been finished before"
+    values::set sdsReplicatedVolume.internal.csiMigrationHook.completed "true"
+    exit 0
+  fi
+
+  echo "Secret ${NAMESPACE}/${SECRET_NAME} does not exist. Starting csi migration"
+
+  sc_list=$(kubectl get sc -o json | jq -r ".items[] | select(.provisioner == \"$old_driver_name\") | .metadata.name")
+  pvc_pv_list=$(kubectl get pv -o json | jq -r --arg oldDriverName "$old_driver_name" '.items[] | select(.spec.csi.driver == $oldDriverName) | .spec.claimRef.namespace + "/" + .spec.claimRef.name + "/" + .metadata.name')
+  pvc_list=$(kubectl get pvc --all-namespaces -o json | jq -r --arg oldDriverName "$old_driver_name" '.items[] | select(.metadata.annotations["volume.kubernetes.io/storage-provisioner"] == $oldDriverName) | .metadata.namespace + "/" + .metadata.name')
+  
+  if [[ -z "$sc_list" && -z "$pvc_pv_list" && -z "$pvc_list" ]]; then
```

Comment:
I would prefer counting the sc/pv/pvc items and put the counts in vars pv_pvc_count/pv_count/sc_count.

Reply:
Got the idea, nevermind.